src/extent.c

```diff
@@ -1225,72 +1222,126 @@ extent_grow_retained(tsdn_t *tsdn, arena_t *arena,
 		*commit = true;
 	}
 
+	/*
+	 * Increment extent_eden_grow_next if doing so wouldn't exceed the legal
+	 * range.
+	 */
+	if (arena->extent_eden_grow_next + egn_skip + 1 < NPSIZES) {
+		arena->extent_eden_grow_next += egn_skip + 1;
+	} else {
+		arena->extent_eden_grow_next = NPSIZES - 1;
+	}
+
+	return extent;
+}
+
+/*
+ * If virtual memory is retained, we allocate from the "eden" region of the
+ * arena.  This creates mappings of exponentially increasing size from which
+ * we can split memory for newly requested extents.  This strategy limits the
+ * total number of disjoint virtual memory ranges retained by each arena, which
+ * both helps merging to be effective, and also helps us avoid any kernel limits
+ * on the number of mappings allowed.
+ */
```

Comment:
nit: explicitly say eden_alloc is the last step and the slow path.

Reply:
Done.